### PR TITLE
PT-875 Trends: Select dates range by clicking on the header

### PR DESCRIPTION
**Pull request by yvon-dblg**
Dependent of: https://github.com/mdblp/viz/pull/38
Marine review: OK.

For the reviewer:
- There is some change in the React flow (like componentWillUpdate -> componentDidUpdate); Theses was deprecated in React 15. They are here to be able to migrate to React 16 one day (and avoid warning message in the console too).
- Some React.PropType where also remove (use PropType directly): deprecated.
- Some unused var was removed too.
- After first review with Marine, we decided to use a double calendar for the range. The code for the single calendar is not remove, as it may be use in the future.

### CODE REVIEW

**Comment by franck-fourel** [comment]
I had to change to use your viz branch to test it: 
```json
"@tidepool/viz": "mdblp/viz#feature/pt876-flexible-trends"
```

File: app/components/datepicker/datepicker.less [hunk 1]
```diff
@@ -152,14 +170,27 @@
 .datepicker-popup-validate {
   display: flex;
   flex-direction: column;
-  margin-top: 0.3em;
+  margin-top: 0.8em;
+  margin-bottom: 0.8em;
 
   .btn {
```
Comment (yvon-dblg) [inline app/components/datepicker/datepicker.less]: Missing &:
`&.btn {`

File: app/components/datepicker/datepicker.less [hunk 2]
```diff
@@ -42,3 +42,155 @@
...
+  border-color: @gray-dark;
+
+  &:hover {
+    color: @white-color !important;
+    font-weight: normal !important;
+  }
+}
+
+/* Range calendar */
+.datepicker-popup-range {
```
Comment (franck-fourel) [inline app/components/datepicker/datepicker.less]: this one may cause issue like that 
image

File: app/components/datepicker/datepicker.js [hunk 3]
```diff
@@ -15,20 +15,313 @@
...
+function containsAll(str, letters) {
+  const ll = letters.length;
+  let ca = true;
+  for (let i=0; i<ll && ca; i++) {
+    ca = ca && str.indexOf(letters[i]) >= 0;
+  }
+  return ca;
+}
+
+class DatePicker extends React.Component {
```
Comment (franck-fourel) [inline app/components/datepicker/datepicker.js]: question about this one: is it the one that was used in the patient page when it creates its own account through the web app?

Comment (yvon-dblg) [reply app/components/datepicker/datepicker.js]: Yes

File: app/components/datepicker/datepicker.less (hunk 2 above)
Comment (yvon-dblg) [reply app/components/datepicker/datepicker.less]: Yes, I need to improve the responsive design.

File: app/components/chart/trends.js [hunk 4]
```diff
@@ -110,17 +105,20 @@ const Trends = translate()(class Trends extends PureComponent {
   }
 
   componentDidMount() {
+    this.log('componentDidMount');
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply app/components/chart/trends.js]: Not really. It won't hurt anyway (not displayed in production mode I think, and need the localStorage key: debug = true).
There is others components with this message, left by Tidepool, It's just useful when you do some debug.
Can be safely removed.

File: app/components/chart/trends.js [hunk 5]
```diff
@@ -110,17 +105,20 @@ const Trends = translate()(class Trends extends PureComponent {
...
     }
   }
 
-  componentWillUnmount = () => {
-    if (this.state.debouncedDateRangeUpdate) {
-      this.state.debouncedDateRangeUpdate.cancel();
-    }
-  };
+  componentDidUpdate() {
+    this.log('componentDidUpdate');
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply app/components/chart/trends.js]: Same reply

File: app/components/chart/trends.js [hunk 6]
```diff
@@ -110,17 +105,20 @@ const Trends = translate()(class Trends extends PureComponent {
...
+  }
+
+  componentWillUnmount() {
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply app/components/chart/trends.js]: Same reply, the previous code in the method is removed now.

File: package.json [hunk 7]
```diff
@@ -41,7 +44,7 @@
     "@babel/polyfill": "7.0.0",
     "@babel/preset-env": "7.1.0",
     "@babel/preset-react": "7.0.0",
-    "@tidepool/viz": "mdblp/viz#dblp.0.9.0",
+    "@tidepool/viz": "mdblp/viz#feature/pt876-flexible-trends",
```
Comment (franck-fourel) [inline package.json]:
```suggestion
    "@tidepool/viz": "mdblp/viz#dblp.0.10.0",
```

**Pull request merged**